Menus/MainMenuGUI.cpp

```diff
@@ -1,3263 +1,539 @@
-					StartAssemblyEditor();
-				} else if (anEvent.GetControl() == m_aEditorButton[GIBEDITOR]) {
-					StartGibEditor();
-				} else if (anEvent.GetControl() == m_aEditorButton[ACTOREDITOR]) {
-					StartActorEditor();
-				}
+		m_MenuScreenChange = false;
+	}
 
-//                g_GUISound.BackButtonPressSound()->Play();
-                g_GUISound.ExitMenuSound()->Play();
-            }
-
-            /////////////////////////////////////////////
-            // META NOTICE SCREEN BUTTONS
-
-			if (m_MenuScreen == METASCREEN)
-            {
-                // Play tutorial button pressed
-                if (anEvent.GetControl() == m_MainMenuButtons[PLAYTUTORIAL])
-                {
-                    // Hide all screens, the appropriate screen will reappear on next update
-                    HideAllScreens();
-                    m_MenuScreen = MAINSCREEN;
-                    m_ScreenChange = true;
-
-                    // Set up and start the tutorial!
-                    g_ActivityMan.SetStartActivity(dynamic_cast<Activity *>(g_PresetMan.GetEntityPreset("GATutorial", "Tutorial Mission")->Clone()));
-					GameActivity * pGameActivity = dynamic_cast<GameActivity *>(g_ActivityMan.GetStartActivity());
-					if (pGameActivity)
-						pGameActivity->SetStartingGold(10000);
-                    g_SceneMan.SetSceneToLoad("Tutorial Bunker");
-                    m_ActivityRestarted = true;
-
-                    g_GUISound.ButtonPressSound()->Play();
-                }
-                // Go to registration dialog button
-                else if (anEvent.GetControl() == m_MainMenuButtons[METACONTINUE])
-                {
-                    m_CampaignStarted = true;
-
-                    // Hide all screens, and stay in main menu for if/when player comes back to the main menu
-                    HideAllScreens();
-                    m_MenuScreen = MAINSCREEN;
-                    m_ScreenChange = true;
-                    g_GUISound.ButtonPressSound()->Play();
-                }
-            }
-
-            /////////////////////////////////////////////
-            // MOD MANAGER SCREEN BUTTONS
-
-			if (m_MenuScreen == MODMANAGERSCREEN)
-            {
-				// Return to main menu button pressed
-				if (anEvent.GetControl() == m_pModManagerBackButton)
-				{
-					// Save settings
-					g_SettingsMan.UpdateSettingsFile();
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-					// Hide all screens, the appropriate screen will reappear on next update
-					HideAllScreens();
-					m_MenuScreen = MAINSCREEN;
-					m_ScreenChange = true;
-					g_GUISound.BackButtonPressSound()->Play();
-				}
+	void MainMenuGUI::ShowCampaignNoticeScreen() {
+		m_MainMenuScreens.at(MenuScreen::CampaignNoticeScreen)->SetVisible(true);
+		m_MainMenuScreens.at(MenuScreen::CampaignNoticeScreen)->GUIPanel::AddChild(m_MainMenuButtons.at(MenuButton::BackToMainButton));
```

Comment:
It doesn't happen at once and all it does is change the parent box.
It can't be child of 2 boxes at once because it removes itself from the old parent in the process.

Reply:
Okay, wasn't sure if gui was set up to remove from old parent and stuff properly. Good stuff then.